tests/ballerina-compiler-api-test/src/test/resources/test-src/reference_type_test.bal

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+type BTable table<map<int>>|BarTable;
+
+type BarTable table<Bar>key(a);
+
+type Bar record {
+    readonly string a;
+};
+
+function testTypeReferenceOfTableConstructor() {
+    BTable tab = table key(a) [{a : "a"}];
+}
```

Comment:
We need to have tests for object scenario.

Reply:
The change is done for `typeInitExpr` but we don't have a node visitor for `typeInitNode`. That's why we skip the test for that.